comp/core/workloadmeta/collectors/internal/remote/processcollector/process_collector_test.go

```diff
@@ -254,11 +270,12 @@ func TestCollection(t *testing.T) {
 
 			time.Sleep(time.Second)
 
+			ctx, cancel := context.WithCancel(context.TODO())
+
 			// remote process collector server (process agent)
-			server := &mockServer{
-				responses:     test.serverResponses,
-				errorResponse: test.errorResponse,
-			}
+			server := newMockServer(ctx, test.serverResponses, test.errorResponse)
```

Comment:
based on my previous comment
this we can do 

```suggestion
			server := newMockServer(context.Background(), test.serverResponses, test.errorResponse)
```